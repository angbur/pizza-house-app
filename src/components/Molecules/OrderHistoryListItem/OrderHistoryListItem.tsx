import Button from 'components/Atoms/Button/Button';
import PriceTag from 'components/Atoms/PriceTag/PriceTag';
import Typography from 'components/Atoms/Typography/Typography';
import { FormType, openDialog } from 'components/Organisms/Dialog/dialogSlice';
import { ThemeContext } from 'components/Theme/ThemeContext';
import { useAppDispatch } from 'hooks';
import { useContext } from 'react';
import { OrderDetail } from 'services/order';
import styled from 'styled-components';
import { OrderItem } from 'types/Order';

const StyledListItem = styled.li`
  border: 2px solid ${(props) => props.theme?.palette.light};
  border-radius: 8px;
  display: flex;
  gap: 3rem;
  justify-content: space-between;
  align-items: center;
  padding: 2rem;
  width: 90%;
  @media (max-width: 950px) {
    gap: 2rem;
    width: auto;
    & > p.paragraph {
      display: none;
    }
  }
  @media (max-width: 460px) {
    & > p {
      display: none;
    }
  }
  @media (max-width: 332px) {
    flex-direction: column;
  }
`;

type OrderHistoryListItemProps = {
  element: OrderDetail;
};

const OrderHistoryListItem = ({ element }: OrderHistoryListItemProps) => {
  const theme = useContext(ThemeContext);
  const dispatch = useAppDispatch();

  const getDescription = (arr: Omit<OrderItem, '_id'>[]) => {
    let result = '';

    arr.map(
      (pizza, index) =>
        (result =
          index != arr.length - 1
            ? result.concat(pizza.item.name, ', ')
            : result.concat(pizza.item.name)),
    );

    return result;
  };

  const handleClick = () => {
    dispatch(openDialog(FormType.orderDetail));
  };

  return (
    <StyledListItem theme={theme}>
      <Typography variant='item-title' color='light'>
        {new Intl.DateTimeFormat('us-US').format(new Date(element.date))}
      </Typography>
      <PriceTag price={element.total} />
      <Typography
        key={`pizza-${element._id}`}
        variant={'paragraph'}
        color='light'
        style={{ textAlign: 'left', width: '360px' }}
      >
        {getDescription(element.order)}
      </Typography>
      <Button variant='primary-light' onClick={handleClick}>
        Details
      </Button>
    </StyledListItem>
  );
};

export default OrderHistoryListItem;
