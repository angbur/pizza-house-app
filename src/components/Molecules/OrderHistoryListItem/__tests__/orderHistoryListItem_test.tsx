import OrderHistoryListItem from '../OrderHistoryListItem';
import { orderHistoryMocks } from 'test/mock/orderHistoryMock';
import { renderWithProviders } from 'test/test-utils';

describe('Order History List item', () => {
  it('should renders order list item correctly', () => {
    const initialState = {
      loggedIn: true,
      status: '200',
      error: undefined,
      userId: 'dummy_id',
      login: 'dummy_login',
      token: 'dummy_token',
    };

    const container = renderWithProviders(<OrderHistoryListItem element={orderHistoryMocks[0]} />, {
      preloadedState: {
        user: initialState,
      },
    }).container;

    expect(container).toMatchSnapshot();
  });
});
