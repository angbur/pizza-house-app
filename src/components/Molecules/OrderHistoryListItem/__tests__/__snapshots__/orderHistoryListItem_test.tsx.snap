// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order History List item should renders order list item correctly 1`] = `
<div>
  <li
    class="sc-jrcTuL ivrLub"
  >
    <h5
      class="item-title"
      style="font-family: Source Sans Pro; font-weight: 600; font-size: 1rem; line-height: 1.4; letter-spacing: 1px; color: rgb(252, 251, 251);"
    >
      13.01.2023
    </h5>
    <p
      class="price-tag"
      style="font-family: Klepon Ijo; font-size: 3.1500000000000004rem; letter-spacing: 0; color: rgb(235, 133, 62);"
    >
      342$
    </p>
    <p
      class="paragraph"
      style="font-family: Source Sans Pro; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(252, 251, 251); text-align: left; width: 360px;"
    >
      Lighthouse
    </p>
    <button
      class="sc-bcXHqe sc-gswNZR feNVH gHullA primary-light btn-lg button"
      style="padding: 0.65rem 1.25rem; font-size: 1.05rem;"
    >
      Details
    </button>
  </li>
</div>
`;
